### PR TITLE
Add source command to CLI

### PR DESCRIPTION
**Pull request by harunurhan**
#5025

A question:
Currently It adds all queries in source file to the history, is this OK or it should add command itself? (like `source /path/to/file`)

### CODE REVIEW

File: presto-cli/src/main/java/com/facebook/presto/cli/Console.java [hunk 1]
```diff
@@ -208,36 +209,23 @@ private static void runConsole(QueryRunner queryRunner, ClientSession session, A
                             System.out.println(getHelpText());
                             continue;
                     }
+
+                    if (command.startsWith("source ")) {
+                        String path = command.split(" ")[1];
+                        String sql = Files.toString(new File(path), UTF_8);
+                        executeCompleteStatements(sql, queryRunner, session, tableNameCompleter, reader);
```
Comment (harunurhan) [inline presto-cli/src/main/java/com/facebook/presto/cli/Console.java]: Should returned partial statement printed like https://github.com/prestodb/presto/blob/master/presto-cli/src/main/java/com/facebook/presto/cli/Console.java#L285 ?

File: presto-cli/src/main/java/com/facebook/presto/cli/Console.java [hunk 2]
```diff
@@ -208,36 +209,23 @@ private static void runConsole(QueryRunner queryRunner, ClientSession session, A
                             System.out.println(getHelpText());
                             continue;
                     }
+
+                    if (command.startsWith("source ")) {
+                        String path = command.split(" ")[1];
```
Comment (harunurhan) [inline presto-cli/src/main/java/com/facebook/presto/cli/Console.java]: Show it be allowed to give multiple paths (space separated)? 
In that case we can do something like:
~~~

foreach path in command.split
    query <- query_from_file(path)
    sql <- sql + query
execute(sql)

Comment (akshatnair) [inline presto-cli/src/main/java/com/facebook/presto/cli/Console.java]: I think for the first pass it is good to just have only one source file.

**Comment by akshatnair** [comment]
Please add some documentation.

**Comment by harunurhan** [comment]
@akshatnai actually I couldn't find a place to add documentation for client commands, other than console help. 
Or did you mean adding it to release docs?

**Comment by akshatnair** [comment]
you can add it in the same section as USE statement

**Comment by cawallin** [comment]
I disagree, USE is SQL and this isn't a SQL feature. There should be a new section for CLI docs, maybe in the "Administration" header.

**Comment by akshatnair** [comment]
@cawallin are you sure "USE" is SQL standard? In the same section we have things for session management and "CALL".

**Comment by cawallin** [comment]
It's not necessarily SQL standard, but it is part of the SQL syntax for Presto. The CLI options are a distinct thing, dealing with how the CLI runs the SQL you give it.

**Comment by harunurhan** [comment]
Although I agree with @cawallin I put docs into sql folder as @akshatnair suggested, since we don't have a separate section for client commands. 

Perhaps we can create another issue about organising docs which will cover this as well.

File: presto-docs/src/main/sphinx/sql/source.rst [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+--------
+
+.. code-block:: none
+
+    SOURCE file-path
+
+Description
+-----------
+
+Execute all complete sql statements in the given file.
```
Comment (akshatnair) [inline presto-docs/src/main/sphinx/sql/source.rst]: Are there any restrictions in the commands that can go in this file? I remember that there is an issue with using "USE" with files and -f option of CLI. Please verify that and add a note to mention such things.

**Comment by akshatnair** [comment]
Small comment in doc otherwise looks good to me.

File: presto-docs/src/main/sphinx/sql/source.rst (hunk 3 above)
Comment (harunurhan) [inline presto-docs/src/main/sphinx/sql/source.rst]: Actually it works as you run all SQL statements in the file by copy/paste-ing to CLI, so USE works too. Since there is no restriction, I didn't add anything to docs. Do you think I should still state that `USE` works?

Comment (akshatnair) [inline presto-docs/src/main/sphinx/sql/source.rst]: Great if there are no restrictions then we do not need to document it.

Comment (electrum) [inline presto-docs/src/main/sphinx/sql/source.rst]: We should document this as part of the CLI docs. I actually don't think `USE` should be documented as a SQL statement, either, since it is only in the CLI (and also shouldn't be part of the main SQL grammar).

File: presto-cli/src/main/java/com/facebook/presto/cli/Console.java [hunk 4]
```diff
@@ -248,6 +236,35 @@ private static void runConsole(QueryRunner queryRunner, ClientSession session, A
         }
     }
 
+    /**
+     * Executes complete statements in {@code sql}
+     *
+     * @return partial statement or empty {@code String}
+     */
+    private static String executeCompleteStatements(String sql, QueryRunner queryRunner, ClientSession session, TableNameCompleter tableNameCompleter, LineReader reader)
```
Comment (electrum) [inline presto-cli/src/main/java/com/facebook/presto/cli/Console.java]: Can you pull this refactoring into a separate commit?

**Comment by harunurhan** [comment]
@electrum I committed refactoring separately. But I didn't do anything about the docs, since I think you may want to do it by organizing all CLI  related docs separately.
